specs/Menu.md

```diff
@@ -1,731 +1,545 @@
+    <MenuList>
+      <MenuItem title="Option 1" />
+      <MenuItem title="Option 2" />
+      <MenuItem title="Option 3" />
+    <MenuList>
+  <CustomSurface>
+)
+```
 
-The menu does not need to provide a native way to be controlled without getting focus directly. Like above, there should be a simple utility wrapper that could be written to implement this.
+### Selection
+
+```typescript
+const trigger = <button> Open menu </button>
+const [selectedItems, setSelectedItems] = React.useState([]);
+
+// basic checkbox example
+const menuCheckbox = (
+  <Menu
+    selectedItems={selectedItems}
+    onSelectionChange={setSeelctedItems}
+    trigger={trigger}
+  >
+    <MenuItem kind="checkbox" index={1} title="Option 1">
+    <MenuItem kind="checkbox" index={2} title="Option 2">
+    <MenuItem kind="checkbox" index={3} title="Option 3">
+  <Menu>
+)
+
+// leverage MenuSection for different selection groups
+const menuSelectableSections = (
+  <Menu
+    selectedItems={selectedItems}
+    onSelectionChange={setSeelctedItems}
+    trigger={trigger}
+  >
+    <MenuSection kind="checkbox">
+      <MenuItem index={1} title="Option 1">
+      <MenuItem index={2} title="Option 2">
+      <MenuItem index={3} title="Option 3">
+    </MenuSection>
+    <MenuSection kind="radio">
+      <MenuItem index={4} title="Option 1">
+      <MenuItem index={5} title="Option 2">
+      <MenuItem index={6} title="Option 3">
+    </MenuSection>
+  <Menu>
+)
+```
 
-##### On Menu Click
+// TODO positioning examples ?
 
-Some implementations of menu have a general `onClick` function that is applied to all items as a way of getting more information.
+## Behaviors
 
-Recommendation: There is not a need to have an `onClick` that is called whenever an item is clicked. Instead the root `menu` should take an `onClick` that will fire if a child's on click does not prevent default.
+### Menu open/dismiss
 
-#### Menu Items
+A menu can be triggered by the following user interactions on the triggering/anchor element. Not all interactions should be supported at the same time, but the component must be able to support combinations of the below interactions.
 
-##### On Item Click
+As a general rule, once the menu is closed the focus should return to the triggering element once the menu is closed unless the interaction would involve another focusable element.
 
-The on item click should be supplied individually by the consuming component
+| Type     | Action     | Result  | Details                                                                |
+| -------- | ---------- | ------- | ---------------------------------------------------------------------- |
+| Mouse    | Click      | Open    | Click on the trigger element                                           |
+| Mouse    | Hover      | Open    | Hover over the trigger element with delay                              |
+| Mouse    | LongPress  | Open    | MouseDown with delay                                                   |
```

Comment:
isn't long press equivalent to right click in touch devices?

Reply:
Yes you're right, I will make the distinction

Reply:
Yes you're right, I will make the distinction